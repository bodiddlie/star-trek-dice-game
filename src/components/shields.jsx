import React from 'react';
import { connect } from 'react-redux';

export const Shields = ({ shields }) =>
  <div>
    {shields}
  </div>;

const mapStateToProps = state => ({
  shields: state.shields,
});

export default connect(mapStateToProps)(Shields);
