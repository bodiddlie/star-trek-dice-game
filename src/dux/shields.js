// @flow
import {
  type Action,
} from './types';

const TAKE_DAMAGE = '[Shields] Take Damage';
const RAISE_SHEILDS = '[Shields] Raise Shields';
const LOWER_SHIELDS = '[Shields] Lower Shields';

export default function reducer(state: number = 0, action: Action): number {
  switch (action.type) {
    case TAKE_DAMAGE: {
      return state - action.payload;
    }
    case RAISE_SHEILDS: {
      return 5;
    }
    case LOWER_SHIELDS: {
      return 0;
    }
    default: {
      return state;
    }
  }
}

export function takeDamage(amount: number): Action {
  return {type: TAKE_DAMAGE, payload: amount}
}

export function raiseShields(): Action {
  return {type: RAISE_SHEILDS, payload: null}
}

export function lowerShields(): Action {
  return {type: LOWER_SHIELDS, payload: null}
}