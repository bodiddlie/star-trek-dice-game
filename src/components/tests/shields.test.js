import React from 'react';
import { shallow } from 'enzyme';

import { Shields } from '../shields';

describe('<Shields />', () => {
  it('should render the shields value', () => {
    const wrapper = shallow(<Shields shields={5} />);
    expect(wrapper).toMatchSnapshot();
  });
});
