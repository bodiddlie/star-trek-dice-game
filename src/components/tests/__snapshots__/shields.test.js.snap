// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Shields /> should render the shields value 1`] = `
<div>
  5
</div>
`;
